wurst/objects/abilities/JumpSystem.wurst

```diff
@@ -9,33 +9,80 @@ import InstantDummyCaster
 import OrderIds
 import Assets
 import ErrorHandling
+import TerrainUtils
+import UnitExtensions
 
-constant let DEFAULT_GRAVITY = 175.
-constant let MAX_TURNSPEED = 3.
-constant let HEIGHT_CHANGE_RATE = 10000.
-constant let CROWFORM_ID = 'Amrf'
+let DEFAULT_GRAVITY = 175.
+let MAX_TURNSPEED = 3.
+let HEIGHT_CHANGE_RATE = 10000.
+let CROWFORM_ID = 'Amrf'
+
+let TARGET_SEARCH_MIN_RANGE = 200.
+let TARGET_SEARCH_INCREMENT = 25
 
 init
     registerCommandAll("jump") (triggerPlayer, args) ->
+        let indexAnim = args.size() > 1 ? args.get(1).toInt() : 2
+        let startinTimeScale = args.size() > 2 ? args.get(2).toReal() : 8.
+        let flyingTimeScale = args.size() > 3 ? args.get(3).toReal() : 0.85
         if (gameConfig.isTestModeEnabled() == true)
             forUnitsSelected(triggerPlayer) u ->
                 if (u.isType(UNIT_TYPE_HERO))
-                    performJump(u, u.getPos().polarOffset(angle(0), 400))
+                    performJump(u, u.getPos().polarOffset(u.getFacingAngle(), 400), indexAnim, startinTimeScale, flyingTimeScale)
         else
             printTimedToPlayer("Requires test mode!", 5., triggerPlayer)
 
-public function performJump(unit u, vec2 targetPos)
+function canJumpToTarget(vec2 targetPos) returns bool
+    return targetPos.isTerrainWalkable()
+
+public function onStartCast(real maxRange)
+    let spellId = GetSpellAbilityId()
+    if spellId != ABILITY_JUMP and spellId != ABILITY_RENDO_ELK_JUMP
```

Comment:
Why does this need to be checked? What other places can call this function?

Reply:
Only EA & Rendo jump call this function, this part is indeed not required anymore.